internal/docstore/memdocstore/mem.go

```diff
@@ -275,6 +283,39 @@ func (c *collection) update(doc map[string]interface{}, mods []driver.Mod) error
 	return nil
 }
 
+// Add two encoded numbers.
+// Since they're encoded, they are either int64 or float64.
+// Allow adding a float to an int, producing a float.
+// TODO(jba): see how other providers handle that.
+func add(x, y interface{}) (interface{}, error) {
+	if x == nil {
+		return y, nil
+	}
+	switch x := x.(type) {
```

Comment:
Is it safe to do
1. convert both to `float64` and calculate.
1. convert back if both are `int64`?

Reply:
Usually, but not always.
https://play.golang.org/p/WAv_3ZENgnx